### ISSUE TITLE
`configOverride.setMergeable(false)` not supported by `ArrayNode`

### ISSUE THREAD

**Issue by ejl888** [FasterXML/jackson-databind#3338]
**Describe the bug**
When updating a JsonNode containing an array. An update will always add the data to the array, even when the
the mapper is configured to not merge Object arrays, ArrayNode, and Collections.

**Version information**
2.13.0

**To Reproduce**

Below a Junit5 test
```
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Collection;

import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;

class ObjectMapperMergeTest {

    @Test
    void itShouldReplaceArrayNodeAndNotAddWhenConfiguredSo() throws IOException {
        // given
        ObjectMapper testSubject = new ObjectMapper();
        testSubject.configOverride(Collection.class).setMergeable(false);
        testSubject.configOverride(ArrayNode.class).setMergeable(false);
        testSubject.configOverride(ArrayList.class).setMergeable(false);
        testSubject.configOverride(Object[].class).setMergeable(false);

        JsonNode mergeTarget = testSubject.readTree("{"
                + "\"array\": [\"Mr.\", \"Ms.\" ],"
                + "\"number\": 888"
                + "}");

        // when
        JsonNode updateNode = testSubject.readTree("{"
                + "\"array\": [\"Mister\", \"Miss\" ]"
                + "}");
        JsonNode patched = testSubject.readerForUpdating(mergeTarget).readValue(updateNode);

        // then
        assertThatJson(patched).node("number").isIntegralNumber().isEqualTo(888);
       // fails because array contains "Mr." and "Ms." also
        assertThatJson(patched).node("array").isArray().containsExactly("Mister", "Miss");
    }
}

```

**Expected behavior**
I would like to have the mapper to replace the array, because now it will never replace it.

**Additional context**
Related issue JsonMerge got unexpected result for List #2315

**Comment by ejl888** [comment]
Fix might add a check for configOverride(ArrayNode.class).setMergeable(false); on line 502 in JsonNodeDeserializer.
If it is set to false, one could set currArray to a new node instead of using the current Node with the existing values or remove all values before starting to add new values.

**Comment by cowtowncoder** [comment]
Thank you for reporting this issue & apologies for slow follow up. This looks like a legit issue, thank you for also providing a reproduction.

**Comment by cowtowncoder** [comment]
Yes, currently this setting is not considered for any `JsonNode` type (and specifically neither for Arrays nor Objects).
So either it would need to be added (and typically introspection should occur when deserializer is constructed, not dynamically), or perhaps added separate as one of new `JsonNodeFeature`s -- if implementation (see https://github.com/FasterXML/jackson-future-ideas/wiki/JSTEP-3) proceeds for 2.14.

**Comment by ejl888** [comment]
@cowtowncoder Adding it to the JsonNodeFeatures sounds ok by me.

**Comment by TsvetanKonstantinov** [comment]
@cowtowncoder Is there a workaround for this as of now?

**Comment by cowtowncoder** [comment]
@TsvetanKonstantinov not that I know of.

**Comment by echatman** [comment]
has anyone found a workaround for this?

**Comment by cowtowncoder** [comment]
Note to self: I think `UntypedObjectDeserializer` shows the way, see `#createContextual()`.
Hoping to look into this more in next few days.

**Comment by cowtowncoder** [comment]
Finally found time to work on this and implemented merge-blocking for upcoming 2.14.0.
